src/Map/src/Bridge/Google/src/GoogleOptions.php

```diff
@@ -0,0 +1,139 @@
+        private bool $fullscreenControl = true,
+        private GoogleOption\FullscreenControlOptions $fullscreenControlOptions = new GoogleOption\FullscreenControlOptions(),
+    ) {
+    }
+
+    public function mapId(?string $mapId): self
+    {
+        $this->mapId = $mapId;
+
+        return $this;
+    }
+
+    public function gestureHandling(GoogleOption\GestureHandling $gestureHandling): self
+    {
+        $this->gestureHandling = $gestureHandling;
+
+        return $this;
+    }
+
+    public function backgroundColor(?string $backgroundColor): self
+    {
+        $this->backgroundColor = $backgroundColor;
+
+        return $this;
+    }
+
+    public function doubleClickZoom(bool $enable = true): self
+    {
+        $this->disableDoubleClickZoom = !$enable;
+
+        return $this;
+    }
+
+    public function zoomControl(bool $enable = true): self
+    {
+        $this->zoomControl = $enable;
+
+        return $this;
+    }
+
+    public function zoomControlOptions(GoogleOption\ZoomControlOptions $zoomControlOptions): self
+    {
+        $this->zoomControlOptions = $zoomControlOptions;
+
+        return $this;
+    }
+
+    public function mapTypeControl(bool $enable = true): self
+    {
+        $this->mapTypeControl = $enable;
+
+        return $this;
+    }
+
+    public function mapTypeControlOptions(GoogleOption\MapTypeControlOptions $mapTypeControlOptions): self
+    {
+        $this->mapTypeControlOptions = $mapTypeControlOptions;
+
+        return $this;
+    }
+
+    public function streetViewControl(bool $enable = true): self
+    {
+        $this->streetViewControl = $enable;
+
+        return $this;
+    }
+
+    public function streetViewControlOptions(GoogleOption\StreetViewControlOptions $streetViewControlOptions): self
+    {
+        $this->streetViewControlOptions = $streetViewControlOptions;
+
+        return $this;
+    }
+
+    public function fullscreenControl(bool $enable = true): self
+    {
+        $this->fullscreenControl = $enable;
+
+        return $this;
+    }
+
+    public function fullscreenControlOptions(GoogleOption\FullscreenControlOptions $fullscreenControlOptions): self
+    {
+        $this->fullscreenControlOptions = $fullscreenControlOptions;
+
+        return $this;
+    }
+
+    public function toArray(): array
+    {
+        return [
+            'mapId' => $this->mapId,
+            'gestureHandling' => $this->gestureHandling->value,
+            'backgroundColor' => $this->backgroundColor,
+            'disableDoubleClickZoom' => $this->disableDoubleClickZoom,
+            'zoomControl' => $this->zoomControl,
+            'zoomControlOptions' => $this->zoomControlOptions->toArray(),
+            'mapTypeControl' => $this->mapTypeControl,
+            'mapTypeControlOptions' => $this->mapTypeControlOptions->toArray(),
+            'streetViewControl' => $this->streetViewControl,
+            'streetViewControlOptions' => $this->streetViewControlOptions->toArray(),
```

Comment:
Maybe we should pass the XXXoptions only when XXX is enabled ? Or enable XXX when XXXoptions(..) is called ? 

(XXX: zoomControl, mapType, streetView, fullscreen)

Reply:
I was thinking something similar yes, in fact we do both! 
Let my know what you think in the last commit :) 
- In the `GoogleOptions`, `*ControlOptions` are passed to the view if their setting is enabled
- In the Stimulus Controller, we assume the user wants to enable (or not) a control if its options are present (or not) in the Map' options